metrics/prometheus.go

```diff
@@ -1568,20 +1568,62 @@ func NewPrometheusCollector(i infoProvider, f ContainerLabelsFunc, includedMetri
 				name:        "container_perf_metric_scaling_ratio",
 				help:        "Perf event metric scaling ratio",
 				valueType:   prometheus.GaugeValue,
-				extraLabels: []string{"cpu", "event"},
+				extraLabels: []string{"cpu", "event", "type"},
 				getValues: func(s *info.ContainerStats) metricValues {
 					values := make(metricValues, 0, len(s.PerfStats))
 					for _, metric := range s.PerfStats {
 						values = append(values, metricValue{
 							value:     metric.ScalingRatio,
-							labels:    []string{strconv.Itoa(metric.Cpu), metric.Name},
+							labels:    []string{strconv.Itoa(metric.Cpu), metric.Name, strconv.FormatUint(uint64(metric.Type), 10)},
 							timestamp: s.Timestamp,
 						})
 					}
 					return values
 				},
 			},
 		}...)
+		machineInfo, err := c.infoProvider.GetMachineInfo()
+		if err != nil {
+			klog.Errorf("Cannot obtain information about machine topology!")
+		} else {
+			c.containerMetrics = append(c.containerMetrics, []containerMetric{
+				{
+					name:        "container_perf_uncore_metric",
+					help:        "Perf uncore event metric.",
+					valueType:   prometheus.CounterValue,
+					extraLabels: []string{"node", "event", "type"},
```

Comment:
regarding the "node" label, perf uncores cannot be collected by "logical" NUMA nodes, it would be more appropriate to name it "socket" label